### DISCUSSION TITLE
Disabling cpu-affinity

### DISCUSSION THREAD

**Discussion by lohash** [RainbowMiner/RainbowMiner#1673] [Q&A] [answered]
Is there a way to either completely disable cpu-affinity for cpu devices globally, or per miner?

If I set “CPUMiningAffinity” to “” in config.txt I get a message on startup that the field is empty or invalid so it will fall back to 0x5.

if I set “Affinity” to “” in specific areas of miners.config.txt it still uses 0x5.

I’ve tried “null” and “0” for those fields as well but no matter what it always falls back to what Rainbow thinks it should be. Can I turn this off?

Thanks!

**Comment by RainbowMiner** [comment]
It can be _set_ per miner - but I am not sure, if I have ever tried disabling it. I'll check this and come back to your question.

**Comment by RainbowMiner** [comment, marked as answer]
Update: in fact, there is no possibility to skip the Affinity/Threads setting for the CPU miners. I'll add a switch per miner to the miners.config.txt. For the meantime, there is a work around:
- find out, which parameters are being used (e.g. `--cpu-affinity` for affinity and `-t` for threads) by looking at the command line, RainbowMiner is using
- edit miners.config.txt and find the miner
- now add the paramters, that you want to have removed to "Params", each followed by a blank and #

e.g.
```
  "Params": "--cpu-affinity # -t #",
```

Using this `#` method _removes_ a certain parameter from the RainbowMiner's miner command line.

**Comment by lohash** [comment]
That worked perfectly, thank you!
  
  
  
    
  

  
  
>
